### DISCUSSION TITLE
[Feature] Stop breaking App <--> Server communication with updates!

### DISCUSSION THREAD

**Discussion by dpantel** [immich-app/immich#11543] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Look, I get that Immich is "under very active development". And I get that this is a good reason not to have docker auto-update the server image -- because shit can break. I accept that I have to read the release notes, and sometimes change my configuration, and go through extra steps when I update my server. And because of this extra work, I don't have time to do it constantly. I do it occasionally, maybe monthly.

What I hate _hate_ **HATE** about Immich is that all of this "active development" gets pushed to my iPhone app ALL. THE. TIME.

So that means that every few days to a couple of weeks, my app can no longer talk to my server. I don't get a notification about this. I just realized I haven't seen a "Memories" in a while, sigh, and go update the server.

This is not a level of function that I will pay money for.

No, I don't want to stop my phone from auto-updating apps. If I could do it app-by-app, I would, but it's all or nothing.

So, I propose 2 possible solutions:

1. Make the app backwards-compatible with older server versions. It doesn't have to be "forever", but for a while, maybe a month or two is reasonable. Then I should get a notification in the app that says "hey, you should go update your server, cause it's getting too old".
2. Somehow make the app update "out of band" (not through the Store). So that the app can track the server version and only update to the latest compatible version.

Alex, you made a pretty app. But even the prettiest... app... gets dumped when it causes too much drama.

### Platform

- [X] Server
- [ ] Web
- [X] Mobile

**Comment by mmomjian** [comment]
Thank you for your feedback. I point you to our disclaimer, which you referenced to:

⚠️ The project is under very active development. Expect bugs and changes. Do not use it as the only way to store your photos and videos!

It is not possible to distribute apps outside the Apple App Store. Making great software requires changes. We are hoping to release a stable version later this year, though this may be delayed.

**Comment by dpantel** [comment]
I made a script to temporize this dumb hole.

https://github.com/dpantel/immich-updater
